<h1> UK ROAD ACCIDENTS </h1>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings

<function warnings.filterwarnings(action, message='', category=<class 'Warning'>, module='', lineno=0, append=False)>

In [2]:
accidents=pd.read_csv('datasets//uk_road_accident.csv')

In [3]:
accidents

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<h3> Descriptive Analytics </h3>

In [4]:
accidents.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [5]:
accidents.describe().T

count       mean       std        min        25%  \
Latitude              660654.0  52.553866  1.406922  49.914430  51.490690   
Longitude             660653.0  -1.431210  1.383330  -7.516225  -2.332291   
Number_of_Casualties  660679.0   1.357040  0.824847   1.000000   1.000000   
Number_of_Vehicles    660679.0   1.831255  0.715269   1.000000   1.000000   

                            50%        75%        max  
Latitude              52.315641  53.453452  60.757544  
Longitude             -1.411667  -0.232869   1.762010  
Number_of_Casualties   1.000000   1.000000  68.000000  
Number_of_Vehicles     2.000000   2.000000  32.000000

<h3> Cleaning the Data Frame </h3>

In [6]:
accidents.isnull()

Index  Accident_Severity  Accident Date  Latitude  Light_Conditions  \
0       False              False          False     False             False   
1       False              False          False     False             False   
2       False              False          False     False             False   
3       False              False          False     False             False   
4       False              False          False     False             False   
...       ...                ...            ...       ...               ...   
660674  False              False          False     False             False   
660675  False              False          False     False             False   
660676  False              False          False     False             False   
660677  False              False          False     False             False   
660678  False              False          False     False             False   

        District Area  Longitude  Number_of_Casualties  Number_of_Vehicles  \
0               False      False                 False               False   
1               False      False                 False               False   
2               False      False                 False               False   
3               False      False                 False               False   
4               False      False                 False               False   
...               ...        ...                   ...                 ...   
660674          False      False                 False               False   
660675          False      False                 False               False   
660676          False      False                 False               False   
660677          False      False                 False               False   
660678          False      False                 False               False   

        Road_Surface_Conditions  Road_Type  Urban_or_Rural_Area  \
0                         False      False                False   
1                         False      False                False   
2                         False       True                False   
3                         False      False                False   
4                         False       True                False   
...                         ...        ...                  ...   
660674                    False      False                False   
660675                    False      False                False   
660676                    False      False                False   
660677                    False      False                False   
660678                    False      False                False   

        Weather_Conditions  Vehicle_Type  
0                    False         False  
1                    False         False  
2                     True         False  
3                    False         False  
4                     True         False  
...                    ...           ...  
660674               False         False  
660675               False         False  
660676               False         False  
660677               False         False  
660678               False         False  

[660679 rows x 14 columns]

In [7]:
accidents.isnull().sum

<bound method DataFrame.sum of         Index  Accident_Severity  Accident Date  Latitude  Light_Conditions  \
0       False              False          False     False             False   
1       False              False          False     False             False   
2       False              False          False     False             False   
3       False              False          False     False             False   
4       False              False          False     False             False   
...       ...                ...            ...       ...               ...   
660674  False              False          False     False             False   
660675  False              False          False     False             False   
660676  False              False          False     False             False   
660677  False              False          False     False             False   
660678  False              False          False     False             False   

        District Are

<h3> Syntax
Data Frame ('column name') </h3>

In [8]:
accidents['Accident_Severity'] == 'Slight'

0         False
1         False
2         False
3         False
4          True
          ...  
660674     True
660675     True
660676     True
660677    False
660678    False
Name: Accident_Severity, Length: 660679, dtype: bool

In [9]:
accidents['Accident_Severity'] == 'Serious'

0          True
1          True
2          True
3          True
4         False
          ...  
660674    False
660675    False
660676    False
660677     True
660678     True
Name: Accident_Severity, Length: 660679, dtype: bool

In [10]:
accidents['District Area'] == 'Kensington and Chelsea'

0          True
1          True
2          True
3          True
4          True
          ...  
660674    False
660675    False
660676    False
660677    False
660678    False
Name: District Area, Length: 660679, dtype: bool

In [11]:
accidents['District Area'] == 'Highland'

0         False
1         False
2         False
3         False
4         False
          ...  
660674     True
660675     True
660676     True
660677     True
660678     True
Name: District Area, Length: 660679, dtype: bool

<h3> How to Change the Data Type of a Column </h3>

In [12]:
accidents.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [13]:
accidents['Index'] = accidents['Index'].astype('category')
accidents['Vehicle_Type'] = accidents['Vehicle_Type'].astype('category')

In [14]:
accidents.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype   
---  ------                   --------------   -----   
 0   Index                    660679 non-null  category
 1   Accident_Severity        660679 non-null  object  
 2   Accident Date            660679 non-null  object  
 3   Latitude                 660654 non-null  float64 
 4   Light_Conditions         660679 non-null  object  
 5   District Area            660679 non-null  object  
 6   Longitude                660653 non-null  float64 
 7   Number_of_Casualties     660679 non-null  int64   
 8   Number_of_Vehicles       660679 non-null  int64   
 9   Road_Surface_Conditions  659953 non-null  object  
 10  Road_Type                656159 non-null  object  
 11  Urban_or_Rural_Area      660664 non-null  object  
 12  Weather_Conditions       646551 non-null  object  
 13  Vehicle_Type             660679 non-null  ca

In [15]:
accidents.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

<h1> Exploratory Data Analytics (EDA) </h1>

<h2>#1.What is the median number of vehicles per accident?

In [16]:
accidents['Number_of_Vehicles'].median()

np.float64(2.0)

<h3> Insight: Most accidents typically involve two vehicles.

<h2>#2.What is the median number of vehicles per accident?

In [17]:
accidents['Number_of_Casualties'].median()

np.float64(1.0)

<h3> Insight: Each accident results in one casualty on average.

<h2>#3. How many accidents involved buses/coaches?

In [18]:
accidents['Vehicle_Type'].str.contains('Bus|Coach', case=False, na=False).sum()

np.int64(27854)

<h3> Insight: A significant number of accidents (27,854 cases) involved buses or coaches. Since buses carry many passengers, these accidents could potentially have higher impact on public safety,

<h2>#4. Which vehicle type is least likely to be in fatal accidents?

In [19]:
accidents.groupby('Vehicle_Type')['Accident_Severity'].value_counts(normalize=True).unstack().fillna(0).sort_values('Fatal').head(5)


C:\Users\Admin\AppData\Local\Temp\ipykernel_19236\4173635683.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accidents.groupby('Vehicle_Type')['Accident_Severity'].value_counts(normalize=True).unstack().fillna(0).sort_values('Fatal').head(5)


Accident_Severity                   Fatal   Serious    Slight
Vehicle_Type                                                 
Data missing or out of range     0.000000  0.000000  1.000000
Ridden horse                     0.000000  0.000000  1.000000
Agricultural vehicle             0.010786  0.144838  0.844376
Goods over 3.5t. and under 7.5t  0.010991  0.140584  0.848425
Taxi/Private hire car            0.011659  0.133218  0.855123

<h3> Insight: involved in fatal accidents these are often vehicles like bicycles, mopeds, or light goods vehicles

<h2> 5.Which district had the highest number of serious accidents?


In [20]:
accidents[accidents['Accident_Severity']=='Serious']['District Area'].value_counts().head(1)

District Area
Birmingham    1474
Name: count, dtype: int64

<h3> insight: Birmingham records the highest number of serious accidents among districts. This could be due to its large population, higher traffic density, or complex road networks.

<h2> #6. Which weather condition has the highest average casualties per accident?

In [21]:
accidents.groupby('Weather_Conditions')['Number_of_Casualties'].mean().sort_values(ascending=False)


Weather_Conditions
Fog or mist              1.452948
Snowing + high winds     1.418079
Raining + high winds     1.416641
Raining no high winds    1.408214
Fine + high winds        1.386018
Other                    1.354869
Fine no high winds       1.350480
Snowing no high winds    1.341776
Name: Number_of_Casualties, dtype: float64

<h3> Insight: Accidents in fog or mist tend to have the highest number of casualties per crash. Poor visibility likely contributes to more severe outcomes.

<h2> #7. How many accidents happened in fog?



In [22]:
accidents['Weather_Conditions'].str.contains('Fog', na=False).sum()

np.int64(3528)

<h3> Insight: Although fog-related accidents (3,528 cases) are relatively fewer compared to clear-weather accidents.

<h2> #8. Which vehicle types are most involved in rural accidents?

In [23]:
accidents[accidents['Urban_or_Rural_Area']=='Rural']['Vehicle_Type'].value_counts().head(5)


Vehicle_Type
Car                                     181922
Van / Goods 3.5 tonnes mgw or under      12156
Bus or coach (17 or more pass seats)      9025
Motorcycle over 500cc                     8957
Goods 7.5 tonnes mgw and over             6156
Name: count, dtype: int64

<h3> Insight: Cars dominate rural accidents, followed by heavy vehicles and larger motorcycles.


<h2> #9.  How many accidents involved motorcycles above 500cc?

In [24]:
accidents['Vehicle_Type'].str.contains('Motorcycle over 500cc', case=False, na=False).sum()

np.int64(25657)

<h3>Insight:Motorcycles above 500cc are a significant contributor to accidents.                </h3>

<h2> #10.Which vehicle type has the highest share of slight accidents?

In [25]:
accidents.groupby('Vehicle_Type')['Accident_Severity'].value_counts(normalize=True).unstack().fillna(0).sort_values('Slight', ascending=False).head()


C:\Users\Admin\AppData\Local\Temp\ipykernel_19236\292165747.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accidents.groupby('Vehicle_Type')['Accident_Severity'].value_counts(normalize=True).unstack().fillna(0).sort_values('Slight', ascending=False).head()


Accident_Severity                        Fatal   Serious    Slight
Vehicle_Type                                                      
Data missing or out of range          0.000000  0.000000  1.000000
Ridden horse                          0.000000  0.000000  1.000000
Bus or coach (17 or more pass seats)  0.012559  0.130342  0.857099
Taxi/Private hire car                 0.011659  0.133218  0.855123
Motorcycle 125cc and under            0.012378  0.133015  0.854607

<h3>Insight:

<h2> #11. Which 10 districts have the most rural accidents?

In [26]:
accidents[accidents['Urban_or_Rural_Area']=='Rural']['District Area'].value_counts().head(10)

District Area
East Riding of Yorkshire    2364
Cornwall                    1992
Highland                    1811
South Cambridgeshire        1803
Aberdeenshire               1781
Leeds                       1774
Powys                       1692
Huntingdonshire             1673
East Lindsey                1538
Carmarthenshire             1527
Name: count, dtype: int64

<h3>Insight: Rural accidents are concentrated in large rural counties (Yorkshire, Cornwall, Highland, Aberdeenshire), which have extensive road networks and high rural traffic.

<h2>#12. What is the total number of accidents in the dataset?

In [27]:
accidents.shape[0]

660679

<h3>Insight: The dataset covers a very large accident record base. This means analysis and trends are statistically reliable, but rare vehicle types or regions (low counts) should be treated cautiously.

<h2>#13. Which 10 vehicle types appear least in the dataset?

In [29]:
accidents['Vehicle_Type'].value_counts().tail(10)


Vehicle_Type
Taxi/Private hire car                    13294
Motorcycle over 125cc and up to 500cc     7656
Motorcycle 50cc and under                 7603
Goods over 3.5t. and under 7.5t           6096
Other vehicle                             5637
Minibus (8 - 16 passenger seats)          1976
Agricultural vehicle                      1947
Pedal cycle                                197
Data missing or out of range                 6
Ridden horse                                 4
Name: count, dtype: int64

<h3>Insight: Taxis, smaller motorcycles, minibuses, and agricultural vehicles are least involved in accidents compared to cars.

<h2>#14. Which 10 districts have the lowest accident counts?

In [36]:
accidents['District Area'].value_counts().tail(10)

District Area
Blaeu Gwent                  222
South Shropshire             220
London Airport (Heathrow)    179
Western Isles                178
Clackmannshire               160
Berwick-upon-Tweed           153
Teesdale                     142
Shetland Islands             133
Orkney Islands               117
Clackmannanshire              91
Name: count, dtype: int64

<h3>Insight: Low accident counts are mostly in remote rural or island areas (Shetland, Orkney, Western Isles), or low-population districts.

<h2>#15. What are the latitude and longitude ranges? 

In [40]:
accidents['Latitude'].min(), accidents['Latitude'].max(), accidents['Longitude'].min(), accidents['Longitude'].max()

(np.float64(49.91443),
 np.float64(60.757544),
 np.float64(-7.516225),
 np.float64(1.76201))

<h3>Insight:

<H2>#16. How many accidents occurred on wet roads vs dry?

In [41]:
accidents['Road_Surface_Conditions'].value_counts(normalize=True)*100

Road_Surface_Conditions
Dry                     67.856499
Wet or damp             28.291106
Frost or ice             2.805806
Snow                     0.892488
Flood over 3cm. deep     0.154102
Name: proportion, dtype: float64

<H3>Insight: Almost 1 in 3 accidents occur on wet/damp roads, highlighting the importance of wet-weather driving conditions.

<h2>#17. What is the distribution of road types?

In [42]:
accidents['Road_Type'].value_counts()

Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

<h3>Insight: Most accidents happen on single carriageways 74%, highlighting them as the most accident-prone road type.

<h2>#18. Which weather condition has the lowest number of accidents?

In [44]:
accidents['Weather_Conditions'].value_counts().tail(1)

Weather_Conditions
Snowing + high winds    885
Name: count, dtype: int64

<h3>Insight:

<h2>#19. Which light condition has the lowest number of accidents?

In [46]:
accidents['Light_Conditions'].value_counts().tail(1)

Light_Conditions
Darkness - lights unlit    2543
Name: count, dtype: int64

<h3>Insight: Accidents in completely unlit darkness are rare compared to daylight or street-lit conditions. However, they are likely to be more dangerous due to poor visibility.

<h2>#20. Which 10 vehicle types appear least often?

In [48]:
accidents['Vehicle_Type'].value_counts().tail(10)

Vehicle_Type
Taxi/Private hire car                    13294
Motorcycle over 125cc and up to 500cc     7656
Motorcycle 50cc and under                 7603
Goods over 3.5t. and under 7.5t           6096
Other vehicle                             5637
Minibus (8 - 16 passenger seats)          1976
Agricultural vehicle                      1947
Pedal cycle                                197
Data missing or out of range                 6
Ridden horse                                 4
Name: count, dtype: int64

<h3>Insight: The rarest vehicle types in the dataset represent specialized or less common transport modes like minibuses, agricultural vehicles, and certain motorcycle categories.

<h1> Date Related 

In [51]:
accidents['Accident Date'] = pd.to_datetime(accidents['Accident Date'], dayfirst=True, errors='coerce')

In [53]:
accidents['Year'] = accidents['Accident Date'].dt.year
accidents['Month'] = accidents['Accident Date'].dt.month
accidents['Day'] = accidents['Accident Date'].dt.day
accidents['DayOfWeek'] = accidents['Accident Date'].dt.dayofweek

In [54]:
accidents

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious           NaT  51.517715   
3       200701BS66128           Serious           NaT  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight           NaT  57.374005   
660675  201091NM01881            Slight           NaT  57.232273   
660676  201091NM01935            Slight           NaT  57.585044   
660677  201091NM01964           Serious           NaT  57.214898   
660678  201091NM02142           Serious           NaT  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type    Year  Month  Day  DayOfWeek  
0                                        Car  2019.0    6.0  5.0        2.0  
1                                        Car  2019.0    7.0  2.0        1.0  
2                      Taxi/Private hire car     NaN    NaN  NaN        NaN  
3       Bus or coach (17 or more pass seats)     NaN    NaN  NaN        NaN  
4                              Other vehicle  2019.0    9.0  3.0        1.0  
...                                      ...     ...

<h2> #21. Which month in 2022 has the most accidents reported?

In [64]:
accidents[accidents['Year'] == 2022 ].groupby(['Month'])['Number_of_Vehicles'].size()

Month
1.0     3243
2.0     5031
3.0     5142
4.0     4209
5.0     4356
6.0     4958
7.0     4989
8.0     4508
9.0     5091
10.0    5459
11.0    5775
12.0    4044
Name: Number_of_Vehicles, dtype: int64

<h3> Insight: 5775 accidents overall, the majority of accidents in 2022 happened in November.

<h2> #22.Trend of accidents per year

In [57]:
accidents.groupby('Year').size()

Year
2019.0    71867
2020.0    70163
2021.0    66172
2022.0    56805
dtype: int64

<H3> Insight:between 2019 and 2022, the total number of accidents fell by 15,062 cases (~21%), showing a consistent year-over-year improvement in safety or reduced road usage.

<h3> #23.In 2020, how many accidents happened during the day?

In [68]:
accidents[(accidents['Light_Conditions'] == 'Daylight' ) & (accidents["Year"] == 2020)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 51605 entries, 182122 to 352702
Data columns (total 18 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   Index                    51605 non-null  category      
 1   Accident_Severity        51605 non-null  object        
 2   Accident Date            51605 non-null  datetime64[ns]
 3   Latitude                 51601 non-null  float64       
 4   Light_Conditions         51605 non-null  object        
 5   District Area            51605 non-null  object        
 6   Longitude                51601 non-null  float64       
 7   Number_of_Casualties     51605 non-null  int64         
 8   Number_of_Vehicles       51605 non-null  int64         
 9   Road_Surface_Conditions  51542 non-null  object        
 10  Road_Type                51176 non-null  object        
 11  Urban_or_Rural_Area      51603 non-null  object        
 12  Weather_Conditions       50425 

<h3> Insight:A total of 51605 accidents occured during daylight in the year 2020

<h2> #24.Which kind of road condition led to more accidents in 2019?

In [69]:
accidents[accidents['Year'] == 2019].groupby(['Road_Surface_Conditions'])['Number_of_Vehicles'].size()


Road_Surface_Conditions
Dry                     52723
Flood over 3cm. deep      102
Frost or ice              845
Snow                      268
Wet or damp             17850
Name: Number_of_Vehicles, dtype: int64

<h3> Insight:A total of 52723 acidents occurred in 2019, with accidents on dry roads occurring often.

<h2> #25.Number of accidents per year

In [63]:
accidents['Year'].value_counts().sort_index()

Year
2019.0    71867
2020.0    70163
2021.0    66172
2022.0    56805
Name: count, dtype: int64

<h3> Insight:In 2019, accidents were the highest at 71,867.